codersdk/workspaceagentconn.go

```diff
@@ -382,6 +385,9 @@ func (c *WorkspaceAgentConn) DebugManifest(ctx context.Context) ([]byte, error)
 		return nil, xerrors.Errorf("do request: %w", err)
 	}
 	defer res.Body.Close()
+	if res.StatusCode != http.StatusOK {
```

Comment:
Ooh, I thought about this in previous PRs but assumed it was handled by `c.apiRequest` 😅